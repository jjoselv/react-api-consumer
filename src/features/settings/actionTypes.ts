export const SET_NATIONALITY = 'SET_NATIONALITY';

export interface SetNationalityAction {
    type: typeof SET_NATIONALITY;
    payload: string;
};
    
export type SETTINGS_ACTION_TYPE = SetNationalityAction;
