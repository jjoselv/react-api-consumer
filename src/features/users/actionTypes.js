export const GET_USERS = 'GET_USERS';
export const GET_NEXT_BATCH = 'GET_NEXT_BATCH';
