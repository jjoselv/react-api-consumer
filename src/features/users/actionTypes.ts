import { User } from "../../components/Users/Users";

export const GET_USERS = 'GET_USERS';
export const GET_NEXT_BATCH = 'GET_NEXT_BATCH';
export type PromiseActions = 'PENDING' | 'FULFILLED' | 'REJECTED';

export interface SetUsersAction {
    type: `${typeof GET_USERS}_${PromiseActions}`;
    payload: {
        data: {
            results: User[];
            info: {
                page: number;
            };
        };
    };
};

export interface SetUsersNextBatchAction {
    type: `${typeof GET_NEXT_BATCH}_${PromiseActions}`;
    payload: {
        data: {
            results: User[];
            info: {
                page: number;
            };
        };
    };
};
    
export type USERS_ACTION_TYPE = SetUsersAction | SetUsersNextBatchAction;
