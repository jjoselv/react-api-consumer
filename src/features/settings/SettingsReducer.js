import config from '../../config';
import {SET_NATIONALITY} from './actionTypes';

const initialState = {
  nationality: config.defaultNationality,
};

function reducer(state = initialState, action) {
  switch (action.type) {
    case SET_NATIONALITY:
      return {
        nationality: action.payload,
      };

    default:
      return state;
  }
};


export default reducer;
