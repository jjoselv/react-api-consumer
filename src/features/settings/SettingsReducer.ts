import config from '../../config';
import {SETTINGS_ACTION_TYPE, SET_NATIONALITY} from './actionTypes';

export interface SettingsState {
  nationality: string
}

const initialState: SettingsState = {
  nationality: config.defaultNationality,
};

function reducer(state = initialState, action: SETTINGS_ACTION_TYPE): SettingsState {
  switch (action.type) {
    case SET_NATIONALITY:
      return {
        nationality: action.payload,
      };
    default:
      return state;
  }
};


export default reducer;
