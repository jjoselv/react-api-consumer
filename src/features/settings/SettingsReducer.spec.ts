import {SET_NATIONALITY} from './actionTypes';
import type {SetNationalityAction, SETTINGS_ACTION_TYPE} from './actionTypes';

import SettingsReducer from './SettingsReducer';

describe('features > settings > SettingsReducer', () => {
  it('returns initial state, if non matched action is dispatched', () => {
    const initialState = {
      nationality: 'ch',
    };

    const action = {
      type: 'FOO',
    };

    expect(SettingsReducer(initialState, action as SETTINGS_ACTION_TYPE)).toBe(initialState);
  });

  it(`updates state according to dispatched action`, () => {
    const initialState = {
      nationality: 'ch',
    };

    const payload = 'es';

    const action: SetNationalityAction = {
      type: SET_NATIONALITY,
      payload,
    };

    expect(SettingsReducer(initialState, action)).toMatchSnapshot();
  });
});
