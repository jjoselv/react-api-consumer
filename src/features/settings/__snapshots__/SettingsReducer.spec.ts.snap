// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features > settings > SettingsReducer updates state according to dispatched action 1`] = `
Object {
  "nationality": "es",
}
`;
